test/unit/tracks/text-track-display.test.js

```diff
@@ -0,0 +1,279 @@
+});
+
+QUnit.test('shows the default caption track first', function(assert) {
+  const player = TestHelpers.makePlayer();
+  const track1 = {
+    kind: 'captions',
+    label: 'English',
+    language: 'en',
+    src: 'en.vtt',
+    default: true
+  };
+  const track2 = {
+    kind: 'captions',
+    label: 'Spanish',
+    language: 'es',
+    src: 'es.vtt'
+  };
+
+  // Add the text tracks
+  player.addRemoteTextTrack(track1);
+  player.addRemoteTextTrack(track2);
+
+  // Make sure the ready handler runs
+  this.clock.tick(1);
+
+  const englishTrack = getTrackByLanguage(player, 'en');
+  const spanishTrack = getTrackByLanguage(player, 'es');
+
+  assert.ok(englishTrack.mode === 'showing', 'English track should be showing');
+  assert.ok(spanishTrack.mode === 'disabled', 'Spanish track should not be showing');
+  player.dispose();
+});
+
+if (!Html5.supportsNativeTextTracks()) {
+  QUnit.test('if user-selected language is unavailable, don\'t pick a track to show', function(assert) {
+    // The video has no default language but has ‘English’ captions only
+    const oldIsFirefox = browser.IS_FIREFOX;
+    const player = TestHelpers.makePlayer();
+    const track1 = {
+      kind: 'captions',
+      label: 'English',
+      language: 'en',
+      src: 'en.vtt'
+    };
+    const captionsButton = player.controlBar.getChild('SubsCapsButton');
+
+    browser.IS_FIREFOX = true;
+    player.src({type: 'video/mp4', src: 'http://google.com'});
+    // manualCleanUp = true by default
+    player.addRemoteTextTrack(track1);
+    // Force 'es' as user-selected track
+    player.cache_.selectedLanguage = 'es';
+
+    this.clock.tick(1);
+    player.play();
+
+    const englishTrack = getTrackByLanguage(player, 'en');
+
+    assert.ok(!captionsButton.hasClass('vjs-hidden'), 'The captions button is shown');
+    assert.ok(englishTrack.mode === 'disabled', 'English track should be disabled');
+
+    browser.IS_FIREFOX = oldIsFirefox;
+    player.dispose();
+  });
+
+  QUnit.test('the user-selected language takes priority over default language', function(assert) {
+    // The video has ‘English’ captions as default, but has ‘Spanish’ captions also
+    const oldIsFirefox = browser.IS_FIREFOX;
+    const player = TestHelpers.makePlayer({techOrder: ['html5']});
+    const track1 = {
+      kind: 'captions',
+      label: 'English',
+      language: 'en',
+      src: 'en.vtt',
+      default: true
+    };
+    const track2 = {
+      kind: 'captions',
+      label: 'Spanish',
+      language: 'es',
+      src: 'es.vtt'
+    };
+
+    browser.IS_FIREFOX = true;
+    player.src({type: 'video/mp4', src: 'http://google.com'});
+    // manualCleanUp = true by default
+    player.addRemoteTextTrack(track1);
+    player.addRemoteTextTrack(track2);
+    // Force 'es' as user-selected track
+    player.cache_.selectedLanguage = 'es';
+    this.clock.tick(1);
+
+    // the spanish captions track should be shown
+    const englishTrack = getTrackByLanguage(player, 'en');
```

Comment:
`player.addRemoteTextTrack` returns an `HTMLTrackElement` (or a shim) that has the track in a `.track` property. That might be another way of getting it rather than `getTrackByLanguage`, if that's the only places that `getTrackByLanguage` is used.